src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/Miscellaneous/MarshalClassTests.cs

```diff
@@ -0,0 +1,188 @@
+        public Exception ex = null;
+
+        [STAThread]
+        [Fact]
+        public void StructWithEnumArray()
+        {
+            TestStructWithEnumArray s = new TestStructWithEnumArray();
+            s.ArrayOfEnum = new TestEnum[3];
+            s.ArrayOfEnum[0] = TestEnum.red;
+            s.ArrayOfEnum[1] = TestEnum.green;
+            s.ArrayOfEnum[2] = TestEnum.blue;
+            int retsize = Marshal.SizeOf(s.GetType());
+            Assert.Equal(12, retsize);
+
+            retsize = 0;
+            retsize = Marshal.SizeOf(typeof(TestStructWithEnumArray));
+            int genericRetsize = Marshal.SizeOf<TestStructWithEnumArray>();
+            Assert.True(genericRetsize == retsize, "Generic and non generic versions of the API did not return the same size!");
+            Assert.Equal(12, retsize);
+        }
+
+        [STAThread]
+        [Fact]
+        public void StructureToPtr()
+        {
+            ip = IntPtr.Zero;
+            Assert.Throws<ArgumentNullException>("ptr", () => Marshal.StructureToPtr<SomeTestStruct>(someTs, ip, true));
+
+            ip = new IntPtr(123);
+            Assert.Throws<ArgumentNullException>("structure", () => Marshal.StructureToPtr<Object>(null, ip, true));
+
+            ip = Marshal.AllocHGlobal(Marshal.SizeOf(someTs));
+            someTs.s = "something";
+            ex = Record.Exception(() => Marshal.StructureToPtr(someTs, ip, false));
+            Assert.Null(ex);
+            ex = Record.Exception(() => Marshal.StructureToPtr(someTs, ip, true));
+            Assert.Null(ex);
+        }
+
+        [STAThread]
+        [Fact]
+        public void PtrToStructure()
+        {
+            ip = IntPtr.Zero;
+            Assert.Throws<ArgumentNullException>("ptr", () => Marshal.PtrToStructure(ip, someTs));
+
+            ip = new IntPtr(123);
+            Assert.Throws<ArgumentNullException>("structureType", () => Marshal.PtrToStructure(ip, null));
+
+            ip = new IntPtr(123);
+            Assert.Throws<ArgumentException>(() => Marshal.PtrToStructure<SomeTestStruct>(ip, someTs));
+        }
+
+        [STAThread]
+        [Fact]
+        public void DestroyStructure()
+        {
+            ip = IntPtr.Zero;
+            Assert.Throws<ArgumentNullException>("ptr", () => Marshal.DestroyStructure<SomeTestStruct>(ip));
+
+            ip = new IntPtr(123);
+            Assert.Throws<ArgumentNullException>("structureType", () => Marshal.DestroyStructure(ip, null));
+        
+            SomeTestStruct_Auto someTs_Auto = new SomeTestStruct_Auto();
+            Assert.Throws<ArgumentException>(() => Marshal.DestroyStructure(ip, someTs_Auto.GetType()));
+
+            ip = Marshal.AllocHGlobal(Marshal.SizeOf(someTs));
+            someTs.s = null;
+            ex = Record.Exception(() => Marshal.StructureToPtr(someTs, ip, false));
+            Assert.Null(ex);
+            ex = Record.Exception(() => Marshal.DestroyStructure<SomeTestStruct>(ip));
+            Assert.Null(ex);
```

Comment:
Same unnecessary-xunit-calls comment; will stop commenting on this.

Reply:
Fixed, see in next commit